### ISSUE TITLE
Migrating from JUnit 4 to JUnit 5

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#876]
Spring has been updated in #734.
Since there are many improvements, divide the stages into several.

### Organize test cases

<details>
<summary>Remove junit-vintage-engine</summary>

 - [x] Resolve the discrepancy between the class name and the test class name
 - [x] Rule rewriting (take time) -> Exclude the rule itself if it is less important to describe it in the rule. If necessary, consider introducing ExternalResource later.(However, it seems that there are many rules that are difficult to handle by injection, such as the need for static.) If migration is possible without using junit-jupiter-migration support, that's a better first landing point. The work of rewriting to a better writing style has a relatively low work risk if it is done after removing junit4.
 - [x] Rewrite necessary annotations
 - [x] Remove junit-vintage-engine

#### memo: `@Rule` Basically deleted. Consider replacing it with an Extension Model or Mock.

 - HomeRule -> Simple rewriting may not be the optimal solution. HomeRule is required for testing that depends on SettingService. The range is very wide, including cases where it depends indirectly, and it has become a barrier when adding test cases. (Probably one reason for poor coverage) Originally, the actual use of SettingService is only necessary for test cases testing SettingService or involving scanning. For other LightWeight dependencies, a better solution maybe simplify the test mechanism.
 - rule that forces parallel iterative execution -> ~~Probably rewritable~~ It's not possible with JUnit5 and it's a pending issue.(Available in cooperation with TestNG, or unrealistic but control with Surefire) This test is rarely run on this project. Therefore, the rules will be deleted as soon as migrate to JUnit5.  The related issue is primarily the sort implementation running in parallel. This issue will be put on hold until have to change the sort implementation, or until Junit 5 supports parallel iterative execution of the case. (Parallel execution is guaranteed if the sort implementation is not altered)Rule removal occurs in PlayQueueTest and JpsonicComparatorsTest.
</details>

<details>
<summary>Use nesting to aggregate test classes in a 1: 1</summary>

Whether nesting is possible depends on the specifications of the test data used.

 - [x] JpsonicComparatorsTest
 - [x] PlaylistServiceTest
 - MediaScannerServiceTest
 - org.airsonic.player.service.search

Scanner Service and Search are on hold for a while. Since it is a heavy mechanism related to Scan, it needs to be redesigned for aggregation. The current mechanism of the scan for test  is not so elegant, but it has the characteristic that only one works at any given time. This was intended to avoid troubles under heavy load on the CI system (Travis). In addition, it does not assume immediate continuous execution, and it is a mechanism to reload the context. It's very slow, but there are few troubles. Integrating multiple test cases for scanning completely different directories is a relatively risky task.　By the way, no one has yet fully proven the integrity of the Sonic server scan.ヾ(o´д｀)人(´д｀o)ﾂ Probably something else to do first.

</details>

<details>
<summary>Speeding up. Consider aggregation of integration test cases on a package-by-package basis</summary>

It is impossible to undertake unless nesting is completely resolved.

</details>

<details>
<summary>Organize Configuration</summary>

 - [x] Replace the rule that previously existed as ``@HomeRule``

</details>

<details>
<summary>Rewrite assertions in a new way(take time)</summary>

It's a concern, but refactoring of already implemented tests is pending for a while. 

 - Add the missing tests. 
 - Newly created tests are written in Junit5.

Because these are more important.
</details>

**Comment by tesshucom** [comment]
LegacyDatabaseStartupTest processing is missing and needs to be fixed.
